storage/range.go

```diff
@@ -1420,6 +1420,15 @@ func (r *Range) splitTrigger(batch engine.Engine, split *proto.SplitTrigger) err
 	}
 	newRng.stats.SetMVCCStats(batch, ms)
 
+	// Insert the range into the RangeTree.
+	rangeTree, err := GetRangeTree(r.rm.Engine(), now)
```

Comment:
This needs to use `batch` instead of `r.rm.Engine()` to be included in the same atomic batch as the split.